rfd/0191-workload-id-config-ux.md

```diff
@@ -0,0 +1,721 @@
+
+In order to use the metadata from the join process as part of rule evaluation
+and templating, a few things need to be adjusted:
+
+1. Upon join, the attested metadata must be converted to the equivelant
+  attributes and persisted in the issued Teleport X509 user certificate.
+2. When issuing Teleport X509 user certificates through role impersonation, the
+  attributes should be propagated to the issued certificate if they exist in the
+  original certificate.
+
+The process of converting attested metadata to attribute key-values will be 
+tailored to each individual join method. This is a suitable place for any 
+adjustments to be made to the values and keys to optimize them for ease of 
+templating and rule evaluation.
+
+The attribute key-value set can be represented as a map, with the key being a
+string and the value being a string.
+
+The attribute key-value set will be encoded as JSON into the X509 certificate
+using a new extension (1.3.9999.2.21).
+
+Whilst out of scope of this RFD, the encoding of these values into the X509
+certificate would enable the inclusion of join metadata within audit logs for
+actions taken using Machine ID, allowing specific actions to be directly linked
+to a specific join without correlating the Bot Instance ID from the audit event
+with a `bot.join` event.
+
+### New SVID Issuance RPC
+
+```proto
+// WorkloadIdentityService provides the signing of workload identity documents.
+service WorkloadIdentityService {
+  rpc IssueWorkloadIdentity(IssueWorkloadIdentityRequest) returns (IssueWorkloadIdentityResponse) {}
+}
+
+message WorkloadIdentityNameSelector {
+  string name = 1;
+}
+
+message WorkloadIdentityLabelSelector {
+  map<string, string> labels = 1;
+}
+
+message WorkloadAttestationAttributes {
+  map<string, string> attributes = 2;
+}
+
+message IssueWorkloadIdentityRequest {
+  // Selector.
+  oneof selector {
+    WorkloadIdentityNameSelector name = 1;
+    WorkloadIdentityLabelSelector labels = 2;
+  }
+  WorkloadAttestationAttributes workload_attributes = 3;
+}
+
+message IssueWorkloadIdentityResponse {
+
+}
+```
+
+When a specific WorkloadIdentity is specified by-name by the client:
+
+1. Fetch WorkloadIdentity resource by name.
+2. Compare `workload_identity_labels` of the client roleset against
+  `metadata.labels` of the WorkloadIdentity. Reject if no access.
+3. Perform rules evaluation of `spec.rules.deny` against the attributes of the
+  client. Reject if no access.
+4. If `spec.rules.allow` is non-zero, complete rules evaluation against the
+  attributes of the client. Reject if no access.
+5. Evaluate templated fields, if attribute included in template missing from
+  attributes of the client, reject.
+6. Issue workload identity credential, emit audit log, and return credential to
+  client.
+
+Where the client has provided label selectors:
+
+1. Fetch all WorkloadIdentity resources.
+2. For each WorkloadIdentity, compare client's requested labels against 
```

Comment:
Thinking about whether or not it makes sense to switch 1 and 2. Should RBAC label matching occur before "requested" label matching? I feel like this question will probably make less sense if/when we build out a label-based cache for WorkloadIdentity resources as at that point, the RBAC labels can be merged with the requested labels.